core/unit_test/TestSharedSpace.cpp

```diff
@@ -156,62 +155,57 @@ TEST(defaultdevicetype, shared_space) {
   Kokkos::fence();
 
   // GET DEFAULT EXECSPACE LOCAL TIMINGS
-  auto defaultExecLocalTimings = incrementInLoop<Kokkos::DefaultExecutionSpace>(
-      defaultExecData, numRepetitions);
+  auto deviceLocalTimings =
+      incrementInLoop<DeviceExecutionSpace>(deviceData, numRepetitions);
 
   // GET DEFAULT HOSTEXECSPACE LOCAL TIMINGS
-  auto defaultHostExecLocalTimings =
-      incrementInLoop<Kokkos::DefaultHostExecutionSpace>(defaultHostExecData,
-                                                         numRepetitions);
+  auto hostLocalTimings =
+      incrementInLoop<HostExecutionSpace>(hostData, numRepetitions);
 
   // GET PAGE MIGRATING TIMINGS DATA
-  std::vector<decltype(defaultExecLocalTimings)> defaultExecSharedTimings{};
-  std::vector<decltype(defaultExecLocalTimings)> defaultHostExecSharedTimings{};
+  std::vector<decltype(deviceLocalTimings)> deviceSharedTimings{};
+  std::vector<decltype(deviceLocalTimings)> hostSharedTimings{};
   for (unsigned i = 0; i < numDeviceHostCycles; ++i) {
     // GET RESULTS DEVICE
-    defaultExecSharedTimings.push_back(
-        incrementInLoop<Kokkos::DefaultExecutionSpace>(sharedData,
-                                                       numRepetitions));
+    deviceSharedTimings.push_back(
+        incrementInLoop<DeviceExecutionSpace>(sharedData, numRepetitions));
 
     // GET RESULTS HOST
-    defaultHostExecSharedTimings.push_back(
-        incrementInLoop<Kokkos::DefaultHostExecutionSpace>(sharedData,
-                                                           numRepetitions));
+    hostSharedTimings.push_back(
+        incrementInLoop<HostExecutionSpace>(sharedData, numRepetitions));
   }
 
   // COMPUTE STATISTICS OF HOST AND DEVICE LOCAL KERNELS
-  auto defaultExecLocalMean     = computeMean(defaultExecLocalTimings);
-  auto defaultHostExecLocalMean = computeMean(defaultHostExecLocalTimings);
+  auto deviceLocalMean = computeMean(deviceLocalTimings);
+  auto hostLocalMean   = computeMean(hostLocalTimings);
 
   // ASSESS RESULTS
   bool fastAsLocalOnRepeatedAccess = true;
 
   for (unsigned cycle = 0; cycle < numDeviceHostCycles; ++cycle) {
-    std::for_each(std::next(defaultExecSharedTimings[cycle].begin()),
-                  defaultExecSharedTimings[cycle].end(),
-                  [&](const uint64_t timing) {
-                    (timing < threshold * defaultExecLocalMean)
+    std::for_each(std::next(deviceSharedTimings[cycle].begin()),
+                  deviceSharedTimings[cycle].end(), [&](const uint64_t timing) {
+                    (timing < threshold * deviceLocalMean)
                         ? fastAsLocalOnRepeatedAccess &= true
                         : fastAsLocalOnRepeatedAccess &= false;
                   });
 
-    std::for_each(std::next(defaultHostExecSharedTimings[cycle].begin()),
-                  defaultHostExecSharedTimings[cycle].end(),
-                  [&](const uint64_t timing) {
-                    (timing < threshold * defaultExecLocalMean)
```

Comment:
So that was the issue?  `defaultExecLocalMean` instead of `hostLocalMean` in the line above?

Reply:
yup ... it was comparing to device timings instead of host timings but host timings is what we wanted